march_hardware/src/IMotionCube.cpp

```diff
@@ -84,9 +84,23 @@ void IMotionCube::mapMosiPDOs()
 }
 
 // Set configuration parameters to the IMC
-void IMotionCube::writeInitialSettings(uint8_t cycle_time)
+bool IMotionCube::writeInitialSettings(uint8_t cycle_time)
 {
-  ROS_DEBUG("IMotionCube::writeInitialSettings");
+  int checksum_verified = verifySetup();
+
+  if (!checksum_verified)
+  {
+    ROS_WARN("The .sw file for slave %d is not equal to the setup of the drive, downloading is necessary",
+             this->slaveIndex);
+    downloadSetupToDrive();
+    checksum_verified = verifySetup();
+    ROS_INFO(checksum_verified ? "writing of the setup data has succeeded" : "writing of the setup data has failed");
+    return true;  // Resets the etherCAT train and imc, necessary after downloading a "new" setup to the drive
```

Comment:
```suggestion
    return true;  // Resets all imcs and restart etherCAT train (necessary after downloading a "new" setup to the drive)
```
Wouldn't it make more sense to return `false` when it fails and `true` when it succeeds?

Reply:
It is coupled to a reset argument, so from that perspective it makes sense to read: my reset argument is True and therefore I must reset.

Reply:
Hmm I believe this is usually done the other way around. You can always do 
`reset = !this->imc_->writeInitialSDOs(cycle_time);`
but you may also keep it as it is, if you prefer.